WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -183,7 +183,12 @@ def update
       end
     else
       flash.now[:danger] = I18n.t('registrations.flash.failed')
-      render :edit
+      if params[:from_admin_view]
+        render :edit
+      else
+        self.register # Set instance variables needed by :register view
```

Comment:
Maybe a "cleaner" way to handle this is to have a private "set_registration_info", that can be shared between `register`, this method, and even `process_payments` (which uses a very similar logic!).

I'd just add a `before_action` for the relevant actions, and directly call the method from here.